math/complex.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * mathematical functions on complex numbers
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <math.h>
+#include <complex.h>
+
+
+double cabs(double complex z)
+{
+	/* FIXME: due to missing hypot() implementation, temporarily sqrt() is used */
```

Comment:
Fun fact - `hypot()` can be implemented via `cabs()` - which should be fundamental then? :)

I guess `hypot()` should be, to not require complex numbers if not needed

Reply:
More than that, however, the `hypot()` function calculate $sqrt(x^2+y^2)$ in such a way that underflow will not occur, and overflow will only occur if the end result deserves it.